steps/Proposals.tsx

```diff
@@ -0,0 +1,609 @@
+};
+
+type FooterProps = {
+    onBackUrl?: string;
+    onBack?: (() => void) | null;
+    backLabel?: string;
+    onCorrect?: (() => void) | null;
+    correctLabel?: string;
+    onForward?: (() => void) | null;
+    forwardLabel?: string;
+    showAccountButton?: boolean;
+    showActionsButton?: boolean;
+    selectVoteAction?: (() => void) | null;
+};
+
+interface VoteBtnProps {
+    backgroundColor: string;
+    children: React.ReactNode;
+    onClick: () => void
+}
+
+const VoteButton = ({ backgroundColor, children, onClick }: VoteBtnProps) => {
+    const VoteBtnsStyle = {
+        height: '50px',
+        width: '70%',
+        backgroundColor: backgroundColor,
+        borderRadius: '30px',
+        margin: '5px',
+        borderStyle: 'none',
+        color: 'white',
+        fontSize: '15px',
+        textAlign: 'left' as const,
+        display: 'flex',
+        alignItems: 'center'
+    }
+
+    return (
+        <button style={VoteBtnsStyle} onClick={onClick} >
+            {children}
+        </button>
+    )
+}
+
+const tableRowStyle = {
+    width: '100%',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+}
+
+interface ProposalData {
+    proposer: string;
+    title: string;
+    description: string;
+}
+
+const Proposals: FC<RequestProposalsProps> = ({ onSuccess, onBack, config, data, header }) => {
+    const [proposals, setProposals] = useState<Proposal[]>([]);
+    const [selected, setSelected] = useState<{ proposalId: number } | null>(null);
+    const [selectedOption, setSelectedOption] = useState<'1' | '2' | '3' | '4'>();
+    const [selectedVoteOption, setSelectedVoteOption] = useState('');
+    const [toggleVoteActions, setToggleVoteActions] = useState(false);
+    const [successHash, setSuccessHash] = useState<string | undefined>();
+    const [loading, setLoading] = useState(true);
+    const [toggleIcon, setToggleIcon] = useState(false);
+    const [filterStatus, setFilterStatus] = useState("all");
+    const { queryClient } = useQueryClient();
+    const { wallet } = useContext(WalletContext);
+    const { chainInfo } = useContext(ChainContext);
+    console.log(filterStatus);
+    // useEffect(() => {
+    //     const castVote = async () => {
+    //         const proposalId = '3';
+    //         const voterAddress = 'ixo1rkyhrz6qz6ydgadwyqjs7cf6ezvz8j2sht0uxg';
+    //         const option = 'VOTE_OPTION_YES';
+    //         const trxMsg = generateVoteTrx({ proposalId, voterAddress, option });
+    //         console.log(trxMsg)
+    //     }
+    //     castVote()
+    // }, [])
+
+    // const handleSelect = async (proposalId: number) => {
+    //     const selectedProposal = proposals.find((proposal) => proposal.proposalId === proposalId);
+    //     if (selectedProposal) {
+    //         setSelectedValue(!selectedValue);
+    //     } else {
+    //         console.log(`${proposalId}`);
+    //     }
+    // }
+
+    const Footer = ({
+        onBack,
```

Comment:
@RafeeqWeideman-YDS Component on its own - cleanup from this file

Reply:
@RafeeqWeideman-YDS Rather try to edit the current footer component to be able to take props to have it accommodate the vote actions based on those props instead of creating a new footer component specifically for the voting.